monasca_persister/repositories/influxdb/abstract_repository.py

```diff
@@ -56,8 +58,12 @@ def _write_batch(self, data_points, database):
                                                    database=database)
                 break
             except influxdb.exceptions.InfluxDBClientError as ex:
-                if (str(ex).startswith(DATABASE_NOT_FOUND_MSG) and
-                        self.conf.influxdb.db_per_tenant):
+                # Expected format: {"error":"database not found: \"test\""}
+                try:
+                    err = jsonutils.loads(ex.content).get('error')
+                except Exception:
```

Comment:
This is bad because if the db was not found and there was some other exception in jsonutils, you would go down the wrong branch and not create the DB and mask an unrelated issue. Putting a default in the get would work.

Reply:
The only exception I saw with jsonutils.loads was that the format of the error was not json or if the json blob doesn't container the 'error' key. This feels safe to me. The scope of exception handling is just for parsing of the exception content.